src/community/geofence-server/src/main/java/org/geoserver/geofence/internal/InternalUserResolver.java

```diff
@@ -42,6 +42,13 @@ public boolean existsUser(String username) {
                     return true;
                 }
             }
+            
+            for (String roleServiceName : securityManager.listRoleServices()) {
+                if (securityManager.loadRoleService(roleServiceName).getRolesForUser(username) != null && 
+                        !securityManager.loadRoleService(roleServiceName).getRolesForUser(username).isEmpty()) {
+                    return true;
+                }
```

Comment:
I would split the user roles retrieve and condition check, I think it will make it more clear and we avoid retrieving two times the same information, something like this:

```
SortedSet<GeoServerRole> userRoles = securityManager.loadRoleService(roleServiceName).getRolesForUser(username);
if (userRoles != null && !userRoles.isEmpty()) {
    return true;
}
```